pkg/storegateway/indexheader/lazy_binary_reader_test.go

```diff
@@ -383,3 +383,37 @@ func TestLazyBinaryReader_ConcurrentLoadingOfSameIndexReader(t *testing.T) {
 		require.Fail(t, "goroutines did not finish in time")
 	}
 }
+
+func TestLazyBinaryReader_SymbolReaderAndUnload(t *testing.T) {
+	t.Parallel()
+	tmpDir, bkt, blockID := initBucketAndBlocksForTest(t)
+
+	testLazyBinaryReader(t, bkt, tmpDir, blockID, func(t *testing.T, r *LazyBinaryReader, err error) {
+		require.NoError(t, err)
+		require.Nil(t, r.reader)
+		t.Cleanup(func() {
+			require.NoError(t, r.Close())
+		})
+
+		closed := atomic.NewBool(false)
+
+		sr, err := r.SymbolsReader()
+		require.NoError(t, err)
+		go func() {
+			<-time.After(1 * time.Second)
+			require.NoError(t, sr.Close())
+
+			closed.Store(true)
+
+			// Multiple close calls should not panic (wg.Done could panic if called too many times).
+			// (It can return error, or not. We don't care).
+			_ = sr.Close()
+			_ = sr.Close()
```

Comment:
There's no guarantee this executes before the test terminates. I would add a WaitGroup to wait for this goroutine to terminate before the test terminates. I understand a panic would get catched anyway, but looks nicer to wait for it.